packages/gitbook-docs/src/markdoc/attributes.ts

```diff
@@ -53,7 +58,34 @@ export class LinkAttr {
             spacePrefix.pathPrefix
           )
         : value;
-    } else return value;
+    } else if (value && isDocsUrl) {
+      const { urlRewrites } = parseContentConfig;
+      const cleanUrl = value.replace(DOCS_URL, '');
+
+      const [currentGuide] = cleanUrl.split('/').filter((p) => p !== '');
+      const correctGuide = urlRewrites[currentGuide] ?? currentGuide;
+      const anchor = cleanUrl.split('#')[1];
+
+      const finalUrl = cleanUrl.replace(currentGuide, correctGuide);
+
+      const guides = parseContentConfig.gitBookPagesWithTitle;
+
+      const guide = guides.find((g) => {
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const [_product, _path, name, _version, ...rest] = g.path
+          .split('/')
+          .filter((p) => p !== '');
+        return finalUrl === `/${name}/${rest.join('/')}`;
```

Comment:
There are too many assumptions about how the URL is formatted; I would perform a simple replace based on the urlRewritesMap to apply the rewrites.